File: C/1.Iniciante/1094.c
#include <stdio.h>

int main(void)
{
    int i, N, amostra, coelho=0, rato=0, sapo=0, total=0;
    char Tipo;

    scanf("%d", &N);

    for(i=1; i<=N; i++){
        scanf("%d %c",&amostra, &Tipo);
        switch(Tipo){
            case 'C':
                coelho = coelho+amostra;
                total = total+amostra;
                break;
            case 'R':
                rato = rato+amostra;
                total = total+amostra;
                break;
            case 'S':
                sapo = sapo+amostra;
                total = total+amostra;
                break;
        }
    }
    printf("Total: %d cobaias\n", total);
    printf("Total de coelhos: %d\n", coelho);
    printf("Total de ratos: %d\n", rato);
    printf("Total de sapos: %d\n", sapo);
    printf("Percentual de coelhos: %.2f %%\n", (((float)coelho)/total)*100);
    printf("Percentual de ratos: %.2f %%\n", (((float)rato)/total)*100);
    printf("Percentual de sapos: %.2f %%\n", (((float)sapo)/total)*100);

    return 0;
}

File: 1041/1094.py
numeros_caso_teste = int(input())

testes_coelho = 0
testes_sapo = 0
testes_rato = 0

testes_total = 0

for i in range(numeros_caso_teste):
    testes = input().split()
    numero_testes = int(testes[0])
    cobaia_teste = testes[1]

    if cobaia_teste == ('C'):
        testes_coelho += numero_testes
    elif cobaia_teste == ('R'):
        testes_rato += numero_testes
    elif cobaia_teste == ('S'):
        testes_sapo += numero_testes

    testes_total += numero_testes

print(f'Total: {testes_total} cobaias')
print(f'Total de coelhos: {testes_coelho}')
print(f'Total de ratos: {testes_rato}')
print(f'Total de sapos: {testes_sapo}')
print(f'Percentual de coelhos: {((testes_coelho / testes_total) * 100):.2f} %')
print(f'Percentual de ratos: {((testes_rato / testes_total) * 100):.2f} %')
print(f'Percentual de sapos: {((testes_sapo / testes_total) * 100):.2f} %')